data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/RetryConfiguration.java

```diff
@@ -78,7 +78,7 @@ public static RetryConfiguration readRetryConfig(final PluginSetting pluginSetti
     if (dlqFile != null) {
       builder = builder.withDlqFile(dlqFile);
     }
-    final Integer maxRetries = (Integer) pluginSetting.getAttributeFromSettings(MAX_RETRIES);
+    final Integer maxRetries = pluginSetting.getIntegerOrDefault(MAX_RETRIES, null);
```

Comment:
Are these part of another PR? This seems unrelated.

Reply:
Casting would not work for secret value substitution especially when it is a String. But it is possible with generic type in PluginConfigValueTranslator this would not be necessary